### PR TITLE
docs: add Intro architecture

### PR DESCRIPTION
**Pull request by omarespejel**
### Description of the Changes

Enhancement: This PR introduces a comprehensive addition to the documentation focused on Starknet's architecture. It provides a detailed overview of the pivotal components within the Starknet ecosystem, including Sequencers, Provers, and nodes.

### PR Preview URL

https://starknet-io.github.io/starknet-docs/pr-1167/documentation/architecture_and_concepts/Network_Architecture/introduction/

### Check List

- [X] Changes have been done against main branch, and PR does not conflict
- [X] PR title follows the convention: `<docs/feat/fix/chore>(optional scope): <description>`, e.g: `fix: minor typos in code`

- - -
This change is [Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/1167)

### CODE REVIEW

**Comment by stoobie** [comment]
Thoughts:
* Which is better: More content on a single page, or smaller chunks of content on multiple pages. I've seen pros and cons for both. How did you decide to make this one large page as opposed to individual pages for Provers, Sequencers, Nodes etc.?
* Given the introductory nature of the Book, and its different style, my gut suggests that we keep it as its own unit within the site. I'm working on a TOC and will share it.

File: components/Starknet/modules/architecture_and_concepts/nav.adoc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 * Architecture
 
-** Blockchain architecture
+** xref:Network_Architecture/introduction.adoc[Starknet Architecture]
```
Comment (stoobie) [inline components/Starknet/modules/architecture_and_concepts/nav.adoc:3]: Logically, this makes sense, but a general UX principle is that it's not a good idea to have more than one function for a given UI element, In this case, I tried to separate out expandable/collapsible sections from content. The idea is that when you click on something, it either expands/collapses, OR it takes you to a page...but not both.
```suggestion
** Starknet architecture
*** xref:Network_Architecture/introduction.adoc[About Starknet architecture]
```

Comment (omarespejel) [reply components/Starknet/modules/architecture_and_concepts/nav.adoc]: This makes a lot of sense! Thanks @stoobie

**Comment by omarespejel** [comment]
What you mention makes sense. I will propose a couple more PRs to better detail the components mentioned: Sequencer, Prover, Nodes.

**Comment by omarespejel** [comment]
@stoobie, if you consider it good. I think we can merge.

File: components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/introduction.adoc [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+:keywords: Starknet, Starknet architecture, Sequencers, Provers, Nodes, Starknet roles, Starknet components, Starknet architecture overview, Starknet architecture introduction, Starknet architecture details, Starknet architecture explanation, Starknet architecture guide, Starknet architecture tutorial, Starknet architecture documentation, Starknet architecture manual, Starknet architecture reference, Starknet architecture handbook, Starknet architecture how-to, Starknet architecture introduction, Starknet architecture overview, Starknet architecture introduction, Starknet architecture details, Starknet architecture explanation, Starknet architecture guide, Starknet architecture tutorial, Starknet architecture documentation, Starknet architecture manual, Starknet architecture reference, Starknet architecture handbook, Starknet architecture how-to
```
[the remark this answers is not in the document]
Comment (stoobie) [reply components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/introduction.adoc:6]: @omarespejel There was a lot of repetition here. Was it intentional?

File: components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/introduction.adoc [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+Starknet’s operation begins when a transaction is received by a gateway,
+which serves as the Mempool. This stage could also be managed by the
+Sequencer. The transaction is initially marked as "RECEIVED." The
+Sequencer then incorporates the transaction into the network state and
+tags it as "ACCEPTED_ON_L2." The final step involves the Prover, which
+executes the operating system on the new block, calculates its proof,
+and submits it to the Layer 1 (L1) for verification.
+
+image::architecture.png[Starknet Architecture,align="center",width="50%"]
```
[the remark this answers is not in the document]
Comment (stoobie) [reply components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/introduction.adoc:25]: And do we really want to reference the same image twice?

**Review by stoobie — CHANGES_REQUESTED** [review]
Next set of suggestions

File: components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/starknet_architecture_overview.adoc [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+
+Nodes that only update their state based on proof validation on L1
+require the least hardware. They have the same requirements as an
+Ethereum node, and once Ethereum light nodes become a reality,
+maintaining such a node could be as simple as using a smartphone. The
+only tradeoff is higher latency. Proofs are sent to Ethereum intermittently, not for every block, resulting in delayed state updates. Plans are in
+place to produce proofs more frequently, even if they are not sent to
+Ethereum immediately, lowering node latency.
+
+// == Conclusion
```
Comment (stoobie) [inline components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/starknet_architecture_overview.adoc:241]: I feel like a Conclusion here is not necessary and doesn't really add any value. In general, it's more appropriate for a Book that you'll read beginning to end, and doesn't fit the style of the docs, at least not where we're placing it.

**Pull request merged**